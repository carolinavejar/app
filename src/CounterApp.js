// Clase 41 - Tarea

import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import M from 'materialize-css';
import "materialize-css/dist/css/materialize.min.css";
import { Button } from 'react-materialize';

const CounterApp = ({valor = 1000}) => {

    const [ counter, setCounter ] = useState( valor );
    const handleAdd = () => {
        setCounter(counter + 1);
    };
    const handleMin = () => {
        setCounter(counter - 1);
    };
    const handleReset = () => {
        setCounter(valor);
    };
    return (
        <>
            <h1>Pruebas contador</h1>
            <h4>{counter}</h4>
            <Button onClick={ handleAdd }> +1 </Button>
            <Button onClick={ handleReset }> Reset </Button>
            <Button onClick={ handleMin }> -1 </Button>           
        </>
       
    );
}

CounterApp.propTypes = {
    valor: PropTypes.number
}

export default CounterApp;