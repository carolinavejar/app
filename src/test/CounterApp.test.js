import React from 'react';
import CounterApp  from '../CounterApp';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';

// No las toma de setupTests.js , así que las dejaré aquí por ahora
import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer( createSerializer ( { mode : 'deep' } ));

describe("Pruebas de Counter App .... .test.js", ()=> { 
    test('debe de mostrar <CounterApp /> correctamente', () => {
        const value = 1000;
        const wrapper = shallow(<CounterApp valor = {value} />);
        expect (wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el valor que se muestra por defecto en valor de 100', () => {
        const wrapper = shallow(<CounterApp valor = { 100 } />);

        const textoParrafo = wrapper.find('h4').text().trim();

        console.log(`texto parrafo ${textoParrafo}`);

        expect (textoParrafo).toBe( `100` );
    })
});

