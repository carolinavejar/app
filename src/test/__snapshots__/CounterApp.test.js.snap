// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Counter App .... .test.js debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Pruebas contador
  </h1>
  <h4>
    1000
  </h4>
  <Button
    node="button"
    onClick={[Function]}
  >
     +1 
  </Button>
  <Button
    node="button"
    onClick={[Function]}
  >
     Reset 
  </Button>
  <Button
    node="button"
    onClick={[Function]}
  >
     -1 
  </Button>
</Fragment>
`;
